src/components/Divider/Divider.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import styled, { StyledComponent } from 'styled-components';
+import { compose, space, SpaceProps } from 'styled-system';
+import { theme } from '../../essentials/theme';
+import { get } from '../../utils/themeGet';
+
+type DividerOffset = number | string;
+
+interface DividerProps extends SpaceProps {
+    /**
+     * Set the direction of the divider to vertical
+     */
+    vertical?: boolean;
+    /**
+     * Set offset / margin of the divider from the surrounding content
+     */
+    offset?: DividerOffset;
+}
+
+const HorizontalLine: StyledComponent<'div', typeof theme, DividerProps, 'theme'> = styled.div.attrs({ theme })<
+    DividerProps
+>`
+    width: 100%;
+    margin: 0 auto;
+    border: 0;
+    border-top: 0.06rem solid ${get('semanticColors.border.primary')};
+
+    ${compose(space)}
+`;
+
+const VerticalLine: StyledComponent<'div', typeof theme, DividerProps, 'theme'> = styled.div.attrs({ theme })<
+    DividerProps
+>`
+    display: inline-block;
+    width: 0.06rem;
+    margin: 0 auto;
+    border: 0;
+    border-left: 0.06rem solid ${get('semanticColors.border.primary')};
+
+    ${compose(space)}
+`;
+
+/**
+ * ** Primary UI element for visually separating content **
+ *
+ * Renders a divider UI component: horizontal or vertical line that visually separates two pieces of data, content or UI immediately next to it.
+ *
+ * Horizontal divider will take up full available width, vertical divider will take up full available height.
+ *
+ * _Divider_ renders a horizontal divider by default. Set **vertical** prop to __true__ to change divider orientation.
+ *
+ *
+ * ---
+ *
+ * <br/>
+ *
+ * #### Divider vs. Border
+ *
+ * The default color of _Divider_ is $border.primary (#C6CDD4) <span style="color: #C6CDD4">⬤</span>, however...
+ *
+ * ** Divider is NOT a border, and should not be used as such. Please do not use this component as a border for elements. **
+ *
+ * Divider is naturally expected to have a certain offset from the elements it is 'dividing' or separating.
+ *
+ * ---
+ *
+ * #### Style Props
+ *
+ * The Divider has following design props:
+ * - **offset** - set the divider offset from the content it is separating (uses _mx_, _my_ styled system props)
+ *
+ */
+const Divider = ({ vertical = false, offset = 0 }: DividerProps): React.ReactElement =>
```

Comment:
@rafael-sepeda would be better to have a default margin as most of the time might be the same or it's going to change and so it makes sense to manually pass it every time?

Reply:
@div-Leo  From my perspective, we should always have default values from props.

Reply:
@rafael-sepeda what would be the default spaceing?

Reply:
@div-Leo default offset (spacing) is 0 right now, so none :)

Reply:
Yep, I know, my suggestion is to have some significant default value

Reply:
No problem for me to set some default offset, idk where we go from to decide what it should be tho ❓ 

I guess we could take it from these two use cases?
<img width="738" alt="Screenshot 2022-05-04 at 15 32 08" src="https://user-images.githubusercontent.com/14894844/166691891-36519d72-39ae-4b3d-9bac-998db5acc3ec.png">

The offset of Dividers is also same as spacing between LabelValue items in the block (in the column flow) = **1rem**

I believe we can safely assume the divider offset will always be same in this kind of block and set default offset to **1rem**.

Reply:
@div-Leo I will set the default spacing to **1rem** then, or do you have other suggestion?